### ISSUE TITLE
Vulkan shows as `Default (Enabled)` in flags, but `brave://gpu` says it is still disabled.

### ISSUE THREAD

**Issue by ghost** [brave/brave-browser#36204]
## Description 
`brave://flags/#enable-vulkan` shows that the default flag's state is Enabled, but when you go to brave://gpu Vulkan shows as disabled.
It will only be enabled when you go to the flag and set it to Enabled.

This is problematic because many phones will have issues if the Vulkan is not enabled, and cause graphical glitches, and unless people don't go to GPU and see that their Vulkan is disabled, they will not try to go to flags and actually enable it to get it enabled.

Maybe there should be a toggle in Settings to quickly enable and disable Vulkan in case some phones have issues with it.

But the problem here is that the flag will say it should be enabled but the default state doesn't reflect that.

## Steps to reproduce 

   1. go to brave://gpu and Vulkan says 'disabled'
   2. Go go brave://flags/#enable-vulkan it should says `Default (Enabled)`
   3. now set the flag to enabled.
   4. Go to GPU page and see how it now says it is enabled. 

## Expected result
GPU page reflect what the flags page says or viceversa.
Either flag = Default (Enabled) = GPU Vulkan Enabled
or GPU Vulkan Disabled = flag Default (Disabled)

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? Yes
- 
## Device details
   - Install type (ARM, x86): ARM
   - Device type (Phone, Tablet, Phablet): Phone
   - Android version: 13
   
## Brave version
1.65.10 Chromium 122.0.6261.43 (Official Build) canary (64-bit)

### Additional information

Someone just had this issue in Stable and had to Enabled the Vulkan manually to stop having glitches. I use Nightly and don't have the issue of glitches, but when I saw the flag vs GPU, I told the person to manually set it to Enabled and apparently it fixed the issues.

So it doesn't affect my phone by having Vulkan on or off, but some people need it, so having the flag doing what it is supposed to be doing should be fixed.

Thank you and have a good day.

**Comment by TEMP-ad** [comment]
Apparently this is not an issue anymore.

**Comment by TEMP-ad** [comment]
@rebron this is still an issue

**Comment by TEMP-ad** [comment]
Vulkan should be used to avoid glitches and saying it is Enabled when it is not Enabled is problematic.

I can't even see the GPU page anymore on android unless it is in Vulkan, but I could copy and paste the text and it shows Vulkan: Disabled, even if the flag says Default (Enabled). With Vulkan Enabled I can see GPU page correctly because the phone breaks without it and doesn't let me see any page just a white screen. 

I thought it was fixed because it's been a long time but no, Vulkan only works if it is set as Enabled and not if it is on Default (Enabled).

**Comment by TEMP-ad** [comment]
@bsclifton Can you do something about this?

After Chromium 127, people are having issues without Vulkan being on, even if the flag says it should be on.

https://www.reddit.com/r/brave_browser/comments/1eeaz1j/youtube_pink_video_issue/

https://www.reddit.com/r/brave_browser/comments/1edtnp9/lecture_vid%C3%A9o_rose/

Like I said in my already deleted comment, I can't even access brave://gpu anymore without Vulkan being on.

This is very important, and it doesn't look like the most difficult fix, which means make sure Vulkan is on by default, if not more issues like this will happen to users. I have a Samsung phone with Android 14, and as you can see affects Samsung tablets with Android 8 are also affected.

I made this issue since February, and now it is affecting users more.

**Comment by TEMP-ad** [comment]
Seems like some people still get the pink video even with Vulkan on, but some people get their video issues fixed with it.

**Comment by bsclifton** [comment]
cc: @SergeyZhukovsky @deeppandya

**Comment by TEMP-ad** [comment]
@SergeyZhukovsky I do have a Samsung A235M with Android 14, it has a Qualcomm Adreno 610 Driver version 0502.39

My problem is not pink screen, but I can't access GPU because it will make the whole content white and I have to restart the browser after that to access anything completely breaking everything.
The problem is that the flag says "Default (Enabled)" but gpu page says disabled, so the only way is to fix it after 127 is force it Enabled. I remember helping someone with glitches by telling to enable Vulkan as well, and at least the OP of the first link (non french one) fixed the issue by enabling Vulkan, so that's why I made this issue, because I found it weird for Vulkan to say enabled in the flags page but not the feature in gpu.

And then when I installed the first nightly of Chromium 127, the blank page was the issue whenever I went to the GPU page. So if Vulkan was enabled as the flags says, my problems would be fixed, so that's what I am reporting here, but if the issue shouldn't happen even without vulkan. fixing any potential GPU issue should be good for everyone.

For example I disabled Vulkan, went to the GPU and copied the information from both with Vulkan Enabled and Disabled, and compared them with VSCode, but also removed the duplicate lines in the Log.

There are not many differences only a:
- Driver Bug Workarounds
  \*   exit_on_context_lost

Initialization time : 64 (vs 74 with Vulkan)

and the vulkan specific informatioon

After that I can only see the same errors being logged over and over again, I added in parenthesis how many times they were logged until the browser even decided to stop reporting more errors. 

```
[ERROR:scoped_egl_image.cc(23)] : Failed to create EGLImage: EGL_BAD_CONTEXT (142)
[ERROR:shared_image_manager.cc(230)] : SharedImageManager::ProduceSkia: Trying to produce a Skia representation from an incompatible backing: EGLImageBacking (119)
[ERROR:raster_decoder.cc(2855)] : [.RenderWorker-0x20055f700]GL ERROR :GL_INVALID_VALUE : glBeginRasterCHROMIUM: passed invalid mailbox. (36)
[ERROR:raster_decoder.cc(3009)] : [.RenderWorker-0x20055f700]GL ERROR :GL_INVALID_OPERATION : glRasterCHROMIUM: RasterCHROMIUM without BeginRasterCHROMIUM (85)
[ERROR:raster_decoder.cc(3098)] : [.RenderWorker-0x20055f700]GL ERROR :GL_INVALID_OPERATION : glEndRasterCHROMIUM: EndRasterCHROMIUM without BeginRasterCHROMIUM (85)
[ERROR:shared_image_representation.cc(487)] : Attempt to read from an uninitialized SharedImage. Initialized region: (0, 0, 0, 0) Size: (1088, 576) (4)
[ERROR:shared_image_representation.cc(487)] : Attempt to read from an uninitialized SharedImage. Initialized region: (0, 0, 0, 0) Size: (544, 576) (477)
[ERROR:raster_decoder.cc(2861)] : [.RenderWorker-0x20055f700]GL ERROR :GL_INVALID_OPERATION : glBeginRasterCHROMIUM: SharedImage not cleared before use. (50)
[ERROR:logger.cc(46)] : Too many GL errors, not reporting any more for this context. use --disable-gl-error-limit to see all errors.
```

**Comment by SergeyZhukovsky** [comment]
@TEMP-ad the flags are considered experimental and they are not intended to be changed a lot by end users. They could be switched on and off dynamically that's what we do with Vulkan. It's off by our variations study for your particular device https://github.com/brave/brave-variations/blob/main/seed/seed.json#L289. And even though you see it as `Default(Enabled)` on your device it's disabled dynamically via the variation study. The flags page doesn't reflect the changes done by variations. So it's not a bug at all, that's how the system designed. The problem is that you can't access the GPU page as you mentioned and some users observe a pink screen. That means that we have to adjust list of devices for that study. But as I mentioned as well as it looks like Vulkan doesn't help everyone and more flags are most likely involved in that.

**Comment by TEMP-ad** [comment]
Well, brave://version says VulkanStudy: Enable, isn't that the one that should say disabled if it was disabled by Brave?

Also, some variations reflect themselves in the flags page on Android, not all, unlike desktop where you see the indication in more flags, but some of the flags enabled or disabled by default with variations are the ones with the asterisk Default (Enabled*) or (Disabled*), Like Speedreader (which doesn't even work since a while but it is Enabled) or CNAME uncloaking (Disabled*) and some ads/rewards ones. 
Just mentioning that because that's what my user experience have taught me, reason why I checked the variations and saw it as Enabled there as well, I ignored it because even if flag didn't have the asterisk, I saw the study as Enabled, so I thought it should have been enabled before.

But if you say there is some weird behavior and disconnection between flags, study and browser, hope it can get fixed soon and then some phones just don't glitchy or weird GPU issues after some major Chromium update.

**Comment by SergeyZhukovsky** [comment]
Yes, the VulkanStudy itself is Enabled and flag is Enabled only if a device doesn't fall under block list parameters https://github.com/brave/brave-variations/blob/main/seed/seed.json#L249.

**Comment by TEMP-ad** [comment]
Tested this with new Nightly and Chromium 128. Problem still presents and I have to force the flag to be on to access brave://gpu.
I see something new though, the grey lines, but this is pretty much what I see when I access GPU page:

![image](https://github.com/user-attachments/assets/480626b1-5264-49f6-beb1-2987bcaae688)

GPU page seems to be the only one affected by this though, I have been running it without Vulkan and nothing weird has came up from that, only GPU being blank and breaking the browser, so as long as I don't visit, Vulkan Enabled or Disabled, doesn't make a difference in normal pages.

**Comment by SergeyZhukovsky** [comment]
@TEMP-ad nothing been done yet to bring Vulkan flag parity with upstream yet, I'm planning to work on it most likely next week.

**Comment by mack00p** [comment]
I experimented a bit and added these arguments `--disable-features=GlobalMediaControls --disable-frame-rate-limit  --disable-gpu-vsync --max-gum-fps="9999"` and now it shows that vulkan is enabled. I hope this helps you.

**Comment by nomadxxxx** [comment]
Any update on this one?
